### PR TITLE
Drop Python 3.5 support, fix CI

### PR DESCRIPTION
**Pull request by ods**
### Changes

* Drop support for Python 3.5
* Revert switch to pytest-asyncio as it doesn't support `unittest.TestCase`
* Remove compatibility code branches for Python versions we already don't support

### Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Review by asvetlov — APPROVED** [review]
LGTM but please wait for a little for @tvoinarovskyi review

**Comment by tvoinarovskyi** [comment]
Hi @ods, thanks for working on this.
Looks good, I am not sure how the ``python-asyncio`` addition did not affect my environment, maybe I failed to rebuild something.

**Comment by ods** [comment]
> I am not sure how the `python-asyncio` addition did not affect my environment, maybe I failed to rebuild something.

`pytest.mark.asyncio` just doesn't work for `unittest.TestCase` methods. I'd like to rewrite tests pytest-style (with fixtures and functions), but that's separate task and deserves a separate PR. And after fixing warnings for Python 3.8/3.9

**Comment by tvoinarovskyi** [comment]
There is a separate branch that removes loop usage called fixpy38 I think. Mostly it's finished aside from a few points that I don't remember any more :)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -4,11 +4,7 @@
 name: Tests
 
 on:
-  push:
```
Comment (tvoinarovskyi) [inline .github/workflows/tests.yml:7]: Feel like we should revert this part. Tests are running 2ce without the limit. I think you use it to run on the fork, right? Maybe there is a better way to configure Actions...

Comment (ods) [reply .github/workflows/tests.yml:7]: Reverted in #669

**Pull request merged**